### ISSUE TITLE
Latest NMI/MIC (1.5.1) compatibility with Key Vault FlexVolume mount

### ISSUE THREAD

**Issue by ankrause** [Azure/aad-pod-identity#336]
**Describe the bug**
There seems to be a backwards compat issue with the Key Vault FlexVolume addon/project under the MSI authentication mode.

Using the latest NMI/MIC (1.5.1) with FlexVolume v0.0.10) I get this error:

> Warning  FailedMount  2m2s  kubelet, k8s-linuxpool1-\<snip\>-vmss000001  MountVolume.SetUp failed for volume "kv-flexvol-0" : mount command failed, status: Failure, reason: /etc/kubernetes/volumeplugins/azure~kv/azurekeyvault-flexvolume failed, F0812 02:34:30.493758 21027 main.go:82] [error] : failed to get keyvaultClient: failed to get key vault token: failed to get service principal token: failed to decode NMI response: json: cannot unmarshal number into Go value of type string

If I use the previous version of MIC/NMI with the same version of the FlexVolume everything works as expected.

**Steps To Reproduce**

1. Deploy a K8S cluster with NMI and MIC versions 1.5.1, and Key Vault FlexVolume 0.0.10
2. Create a deployment that mounts a volume using MSI authentication.

Sample identity:
```
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentity
metadata:
  name: azure-identity
spec:
  type: 0
  ResourceID: {ResourceId}
  ClientID: {ClientId}
---
apiVersion: aadpodidentity.k8s.io/v1
kind: AzureIdentityBinding
metadata:
  name: azure-identity-binding
spec:
  AzureIdentity: azure-identity
  Selector: MSI
```

Sample volume mount:
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    app: {Name}
  name: {Name}
spec:
  replicas: 1
  selector:
    matchLabels:
      app: {Name}
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 10%
    type: RollingUpdate
  template:
    metadata:
      labels:
        aadpodidbinding: MSI
        app: {Name}
    spec:
      containers:
        image: {Image}
        imagePullPolicy: IfNotPresent
        name: {Name}
        volumeMounts:
        - mountPath: /run/certs
          name: kv-flexvol-0
          readOnly: true
      nodeSelector:
        beta.kubernetes.io/os: linux
      volumes:
      - flexVolume:
          driver: azure/kv
          options:
            keyvaultname: {VaultName}
            keyvaultobjectnames: {SecretName}
            keyvaultobjecttypes: secret
            resourcegroup: {ResourceGroupName}
            subscriptionid: {SubscriptionId}
            tenantid: {TenantId}
            usepodidentity: "true"
        name: kv-flexvol-0
```

**Expected behavior**
Identity is assigned to the pod and Key Vault FlexVolume correctly mounts the designated volume using MSI authentication.

**AAD Pod Identity version**
NMI/MIC 1.5.1

**Kubernetes version**
1.14.5

**Additional context**
Not sure who would own this issue, but it looks like I can't upgrade to the latest pod identity images yet to get all the fixes and improvements.

I've also tried using this with v0.0.11 of the FlexVolume, but I have (different) problems combining those together.

**Comment by kkmsft** [comment]
@ankrause - 

Thank you for trying out 1.5.1. 

As aad-pod-identity moved to new adal libraries, the usage of older libraries had become incompatible (due to backward combat issues introduced with Azure/go-autorest#326). With the PR #306 backward compatibility was added in paths where applications fetch tokens. The host handler path which is used by kevvault flex volume also needs similar fixes. PR #337 should fix this issue. I will test this on Monday morning and work towards merging this. 

>>Not sure who would own this issue, but it looks like I can't upgrade to the latest pod identity images yet to get all the fixes and improvements.

@ankrause - Thank you for raising the issue here. Between me and @aramase, we will get it sorted by either using the PR #337 or upgrading the adal libraries in key vault flex volume, which ever is faster after testing out the hypothesis.

**Comment by kkmsft** [comment]
Fixed by #337
